src/target_intacct/__init__.py

```diff
@@ -44,6 +46,8 @@ def main() -> None:
         statistical_journal_upload(intacct_client, object_name, batch_title)
     elif object_name == "employee_rate":
         employee_rate_upload(intacct_client)
+    elif object_name == "payment_record":
```

Comment:
this whole chain of if/elif could be converted to a Python match if python 3.10 is supported.

Another option is a dictionary that maps the object_name to the function it will call, then do a dictionary get, call the retrieved function. 

Minor -- just mentioning those options for completeness.